pymeasure/instruments/common_base.py

```diff
@@ -408,6 +424,10 @@ class SpecificInstrument(GenericInstrument):
         parameters name except `dynamic` and `docs` (e.g. `values` in the example) has to be
         considered reserved for dynamic property control.
         """
+        if kwargs:
+            warn(f"Do not use keyword arguments {kwargs} as `control` parameter "
+                 f"for the `values` method, use `v_kwargs` parameter instead. docs:\n{docs}",
+                 FutureWarning)
 
```

Comment:
at this point you could merge/update any present kwargs into `v_kwargs` and continue, then below in the values and other calls you can already use the "new" API without **kwargs, and have less adjustment work later when removing the deprecated code.

Reply:
It does not work (which caused major headaches with the tests... ), as the "kwargs" dict lives in the property creator.
So, when I add something to it in the first property, all other properties have that same argument...

Reply:
I tried to do after the warning:
~~~python
v_kwargs.update(kwargs)
~~~

In the tests, at one point, I verify the kwarg `testing=True`. All following tests complain, that they receive an illegal argument `testing`.

My only idea is, that I change the default v_kwarg for all following usages of that static method.

I could try a temporary variable:
~~~
kw_temp = {}
if kwargs:
    warn()
    kw_temp.update(kwargs)
kw_temp.update(v_kwargs)
...
vals = self.values(..., **kw_temp)
~~~

Reply:
This is probably an effect of having these "closures" around, this is useful but can be tricky at times. 
I don't immediately see where the problem comes from, but we should be able to avoid it cleanly. 
I'll probably have to pull the branch to troubleshoot.

Reply:
No wait, you have `v_kwargs={}` in the signature! This is a mutable default argument, and as such a [big no-no](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) in Python. Probably this is causing the problem, and even if not it must be avoided. Can you avoid that and re-check? E.g. use `None` and `if values_kwargs is None: values_kwargs = {}` (or other solution).

Reply:
That is the problem (very probably). 
Thanks for finding it. Again I learned something new (or understood it better, why something with lambda failed me in the past).

Reply:
Mutable default args are not related to lambdas. 
What I think you are thinking of is lambdas and closures: see e.g. https://realpython.com/python-lambda/#closure and https://realpython.com/inner-functions-what-are-they-good-for/#retaining-state-with-inner-functions-closures ?

Reply:
I know, that they are two different things, but they are kind of related and both appeared in the link you gave me.

Reply:
I do not remember exactly the problem, but it was something like mutable lambda as function argument.
That did not work out...

Therefore the connection.